packages/flame/lib/src/components/component.dart

```diff
@@ -93,10 +190,45 @@ abstract class Component {
   ///   myImage = await gameRef.load('my_image.png');
   /// }
   /// ```
-  Future<void>? onLoad() => null;
+  Future<void> onLoad() async {}
```

Comment:
why not `Future<void>?`?

Reply:
The `GameWidget` was not happy about that, I didn't look any further into why that was.
Is it a lot more expensive to do a function call than to do a null check?

Reply:
When the `prepare()` function calls `component.onLoad()`, it checks if the return value is `null` and in that case avoids `await`-ing it.  Thus, any component that doesn't override `onLoad` can be added slightly faster.

Reply:
Reverted it.